src/Symfony/Component/DependencyInjection/ContainerBuilder.php

```diff
@@ -321,6 +321,11 @@ public function getReflectionClass(?string $class, bool $throw = true): ?\Reflec
         if (!$class = $this->getParameterBag()->resolveValue($class)) {
             return null;
         }
+
+        if ($this->isInternalType($class)) {
```

Comment:
I'd suggest to use a private static array instead, and put types as keys in the array, so that checking could be done inline with a simple isset()

Reply:
Oh, you are right. It would be much better.